symengine/pow.cpp

```diff
@@ -94,7 +94,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero) && is_a_Number(*b)) {
```

Comment:
That's what's happening in `x/x` as well if `x` is zero or `0/x` if `x` is zero.
I think since we do that already, this is acceptable too.